docs/nx-cloud/reference/custom-steps.md

```diff
@@ -0,0 +1,101 @@
+
+- **Create a Dedicated Repository**: You can create a separate repository that contains a collection of custom steps. This can be useful for sharing steps across multiple projects or teams.
+- **Same Repository as your Nx Workspace**: You can place the custom step file in the same repository as your Nx workspace. Ensure that the custom step file is publicly accessible on GitHub.
+
+{% /callout %}
+
+2. **Define the Step Template**: Below is an example of a custom step definition:
+
+```yaml {% fileName=".nx/workflows/custom-step.yaml" %}
+name: 'Custom Step'
+description: 'This is a custom step that does XYZ.'
+definition:
+    using: 'node'
+    main: './scripts/custom-script.js'
+    post: './scripts/post-custom-script.js'
+inputs:
+    - name: 'input1'
+    description: 'Description for input1'
+    default: 'default_value'
+    required: true
+    - name: 'input2'
+    description: 'Description for input2'
+    required: false
+```
+
+### Explanation of Fields
+
+- **name**:
+
+  - The name of the custom step.
+
+- **description**:
+
+  - A description of what the custom step does.
+
+- **definition**:
+
+  - **using**: The runtime environment. Supported values are `node` and `aggregate`.
+    - **node**: If `using` is set to `node`, then only `main` and `post` can be used.
+    - **aggregate**: If `using` is set to `aggregate`, then only `steps` can be used.
+  - **main**: Path to the main script to run (only if `using` is `node`).
+  - **post**: Path to the post script to run (only if `using` is `node`).
+  - **steps**: A list of sub-steps to be executed (only if `using` is `aggregate`). Steps follow the same definition as [launch templates](/ci/reference/launch-templates#launch-template-structure).
+
+- **inputs**:
+  - **name**: The name of the input.
+  - **description**: A description of the input.
+  - **default**: Default value for the input.
+  - **required**: Whether the input is required.
+
+### Using Inputs in Scripts
+
+If your custom step has inputs, they can be accessed in scripts or JavaScript files using environment variables. Each input is prefixed with `NX_CLOUD_INPUT_`. For example, if you have an input called `input1`, you can access it in a JavaScript file like this:
+
+```javascript
+const input1 = process.env.NX_CLOUD_INPUT_input1;
+console.log(`The value of input1 is: ${input1}`);
+```
+
+### Using Custom Steps in Launch Templates
+
+Once you've defined your custom step, you can use it in your launch templates by referencing the custom step file. Here's an example:
+
+```yaml {% fileName=".nx/workflows/agents.yaml" %}
+launch-templates:
+  custom-template:
+    resource-class: 'docker_linux_amd64/medium'
+    image: 'ubuntu22.04-node20.11-v7'
+    init-steps:
+      - name: Custom Step
+        uses: 'your-org/custom-steps-repo/.nx/workflows/custom-steps.yaml'
+        env:
+          CUSTOM_VAR: 'custom_value'
```

Comment:
it might be better to not give an example of `env:` as it might confuse people on how it's different from inputs

`env` is not that useful anymore now that we have inputs?

Reply:
I think having env shown is good, because there could be cases where someone just wants a regular env without having to deal with `NX_CLOUD_INPUT_`. I can add more details that using inputs would enable better validation though.